### PR TITLE
add dist of rec model

### PR DESCRIPTION
**Pull request by littletomatodonkey**

### CODE REVIEW

**Review by weisy11 — DISMISSED** [review]
LGTM

**Review by RainFrost1 — DISMISSED** [review]

**Review by TingquanGao — APPROVED** [review]

File: ppcls/configs/GeneralRecognition/GeneralRecognition_PPLCNet_x2_5_dml.yaml [hunk 1]
```diff
@@ -0,0 +1,194 @@
+# global configs
+# global configs
```
Comment (TingquanGao) [inline ppcls/configs/GeneralRecognition/GeneralRecognition_PPLCNet_x2_5_dml.yaml:2]: 多了一行。。。

File: ppcls/engine/evaluation/classification.py [hunk 2]
```diff
@@ -78,10 +78,10 @@ def classification_eval(engine, epoch_id=0):
...
                 if isinstance(out, dict):
-                    if "logits" in out:
-                        out = out["logits"]
-                    elif "Student" in out:
+                    if "Student" in out:
                         out = out["Student"]
+                    elif "logits" in out:
+                        out = out["logits"]
                     else:
                         msg = "Error: Wrong key in out!"
```
Comment (TingquanGao) [inline ppcls/engine/evaluation/classification.py:86]: 是不是需要加一个：
logger.error(msg)
我当时好像漏掉了

File: ppcls/configs/GeneralRecognition/GeneralRecognition_PPLCNet_x2_5_dml.yaml [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    - DistillationDMLLoss:
+        weight: 1.0
+        key: "logits"
+        model_name_pairs:
+        - ["Student", "Teacher"]
+    - DistillationDMLLoss:
+        weight: 1.0
+        key: "logits"
+        model_name_pairs:
+        - ["Student", "Teacher"]
```
Comment (TingquanGao) [inline ppcls/configs/GeneralRecognition/GeneralRecognition_PPLCNet_x2_5_dml.yaml:92]: 为什么这里有两个相同的定义，这样正确么？

File: ppcls/configs/GeneralRecognition/GeneralRecognition_PPLCNet_x2_5_dml.yaml [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    coeff: 0.00001
+
+
+# data loader for train and eval
+DataLoader:
+  Train:
+    dataset:
+      name: ImageNetDataset
+      image_root: ./dataset/
+      cls_label_path: ./dataset/train_reg_all_data.txt
```
Comment (TingquanGao) [inline ppcls/configs/GeneralRecognition/GeneralRecognition_PPLCNet_x2_5_dml.yaml:116]: 这里的路径是否需要配置成我们实际使用的数据集的路径？

File: ppcls/configs/GeneralRecognition/GeneralRecognition_PPLCNet_x2_5_udml.yaml [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+    coeff: 0.00001
+
+
+# data loader for train and eval
+DataLoader:
+  Train:
+    dataset:
+      name: ImageNetDataset
+      image_root: ./dataset/
+      cls_label_path: ./dataset/train_reg_all_data.txt
```
Comment (TingquanGao) [inline ppcls/configs/GeneralRecognition/GeneralRecognition_PPLCNet_x2_5_udml.yaml:115]: 这里是否需要配置为我们实际使用的数据集路径

File: ppcls/configs/ImageNet/Distillation/mv3_large_x1_0_distill_mv3_small_x1_0.yaml [hunk 6]
```diff
@@ -85,7 +87,7 @@ DataLoader:
 
     sampler:
         name: DistributedBatchSampler
-        batch_size: 512
+        batch_size: 256
```
Comment (TingquanGao) [inline ppcls/configs/ImageNet/Distillation/mv3_large_x1_0_distill_mv3_small_x1_0.yaml:90]: bs修改，学习率需要改么？

File: ppcls/loss/distillationloss.py [hunk 7]
```diff
@@ -68,7 +69,7 @@ def __init__(self,
 
     def forward(self, predicts, batch):
         loss_dict = dict()
-        for idx, name in enumerate(self.model_names):
+        for _, name in enumerate(self.model_names):
```
Comment (TingquanGao) [inline ppcls/loss/distillationloss.py:72]: idx 不需要的话，是不是可以直接写：
for name in self.model_names:

File: ppcls/loss/dmlloss.py [hunk 8]
```diff
@@ -32,15 +32,19 @@ def __init__(self, act="softmax"):
...
 
-        log_out1 = paddle.log(out1)
-        log_out2 = paddle.log(out2)
-        loss = (F.kl_div(
-            log_out1, out2, reduction='batchmean') + F.kl_div(
-                log_out2, out1, reduction='batchmean')) / 2.0
+    def forward(self, x, target):
+        if self.act is not None:
+            x = F.softmax(x)
+            target = F.softmax(target)
```
Comment (TingquanGao) [inline ppcls/loss/dmlloss.py:46]: 这里似乎应该是这样？

x = self.act(x)
target = self.act(target)
